### PR TITLE
feat (#19407 JAVA SPRING WEBCLIENT): Add support for useSingleRequestParameter to Spring WebClient

### LINKED ISSUE #19407: [REQ][Java][Client][WebClient] Group Parameters as one parameter (useSingleRequestParameter) for Spring WebClient
Like for java native, okhttp and jersey add the possibility to group all parameters into one (object) when `useSingleRequestParameter ` is set `true`.

## Describe the solution you'd like

A class containing all parameters which is generated combined with a changed method heard when the option is set should do the trick. :)

- [ ] And, of course, don't forget to update the options documentation when done. 

## Describe alternatives you've considered

/

## Additional context

Example Solutions:
https://github.com/OpenAPITools/openapi-generator/blob/eda4547f151a7b5a8952387fa1cb7974c0a5e5ee/modules/openapi-generator/src/main/resources/Java/libraries/microprofile/api.mustache

https://github.com/OpenAPITools/openapi-generator/blob/eda4547f151a7b5a8952387fa1cb7974c0a5e5ee/modules/openapi-generator/src/main/resources/Java/libraries/jersey3/api.mustache

(I'll create a PR soon. This issue is mainly for me to don't forget it ^^)

### PR DESCRIPTION
**Pull request by Nicklas2751**
As Requested with #19407 I added support for useSingleRequestParameter to the Spring WebClient

Closes #19407 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming `7.x.0` minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @martin-mfg (2023/08)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache [hunk 1]
```diff
@@ -37,23 +37,96 @@ public class {{classname}} {
     private ApiClient apiClient;
 
     public {{classname}}() {
-        this(new ApiClient());
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache:40]: thanks for the PR

can we keep 4-space as the indention to make it consistent with the rest of the file?

**Comment by Nicklas2751** [comment]
1 check failed because I rebased my branch with the current master branch which triggered the (not working) scala action

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/11280772589/job/31374692898?pr=19827

please update the samples when you've time

File: modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache [hunk 2]
```diff
@@ -46,14 +46,81 @@ public class {{classname}} {
     }
 
     public ApiClient getApiClient() {
-        return apiClient;
+       return apiClient;
```
Comment (martin-mfg) [inline modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache:49]: Not important, but why remove the space here? It seemed good before.

Comment (Nicklas2751) [reply modules/openapi-generator/src/main/resources/Java/libraries/webclient/api.mustache:49]: oh, that wasn't intentional

**Comment by Nicklas2751** [comment]
@wing328  This is done. I also rebased with the latest upstream master :)

**Comment by wing328** [comment]
Sorry for the delay. Just merged the PR.

Thanks again for your contribution.

**Pull request merged**